contracts/PetPark.sol

```diff
@@ -3,5 +3,94 @@ pragma solidity ^0.8.0;
 
 
 contract PetPark {
+    address public owner;
+
+    struct Borrower{
+        bool isBorrowed;
+        uint petType;
+        uint gender;
+        uint age;
+    }
+
+    struct Pet{
+        bool isAvailable;
+        uint number;
+    }
+
+    mapping(uint => Pet) pets;
+    mapping(address => Borrower ) listBorrower;
+
+    constructor(){
+        // Set the transaction sender as the owner of the contract
+        owner = msg.sender;
+    }
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Not an owner");
+        _;
+    }
+
+    // Add a pet to the pet park
+    event Added(uint petType, uint petCount);
+
+    //Emit event when a pet is borrowed
+    event Borrowed(uint petType);
+
+
+    function add(uint _animalType, uint _count) onlyOwner public {
+        require(_animalType > 0 && _animalType <= 5 , "Invalid animal");
+
+        // Add the number of pets of the given type to the pets mapping
+        pets[_animalType] = Pet({number : _count + pets[_animalType].number, isAvailable: true });
+
+        emit Added(_animalType, pets[_animalType].number);
+    }
+
+    function borrow(uint _age, uint _gender, uint _animalType)  public{
+        require(_age > 0, "Invalid Age");
+
+        require(_animalType > 0 && _animalType <= 5 , "Invalid animal type");
+        
+        require(pets[_animalType].isAvailable, "Selected animal not available");
+
+        if(listBorrower[msg.sender].isBorrowed){
```

Comment:
Since `listBorrower[msg.sender]` is being used multiple times it can be stored in a single memory variable. Later in the course you will learn that it is a gas optimisation technique